devlib/target.py

```diff
@@ -1821,6 +1823,138 @@ def is_not_set(self, name):
     def has(self, name):
         return self.get(name) in ['m', 'y']
 
+class KernelConfigTristate(Enum):
+    YES = 'y'
+    NO = 'n'
+    MODULE = 'm'
+
+    def __bool__(self):
+        """
+        Allow using this enum to represent bool Kconfig type, although it is
+        technically different from tristate.
+        """
+        return self in (self.YES, self.MODULE)
+
+    def __nonzero__(self):
+        """
+        For Python 2.x compatibility.
+        """
+        return self.__bool__()
+
+    @classmethod
+    def from_str(cls, str_):
+        for state in cls:
+            if state.value == str_:
+                return state
+        raise ValueError('No kernel config tristate value matches "{}"'.format(str_))
+
+class TypedKernelConfig(KernelConfigBase):
+    """
+    Mapping-like typed version of :class:`KernelConfig`.
+
+    Values are either :class:`str`, :class:`KernelConfigTristate`, or
+    :class:`int`. ``hex`` Kconfig type is mapped to :class:`int` and ``bool``
+    to :class:`KernelConfigTristate`.
+
+    The equivalent features of :class:`KernelConfig`, fitting in the Mapping
+    API are:
+
+        * ``if kernel_config.is_enabled('FOO')``: ``if kernel_config.get('FOO')``
+          (strictly: ``if kernel_config.get('FOO') is KernelConfigTristate.YES``)
+
+        * ``if kernel_config.has('FOO')`` : ``if kernel_config.get('FOO')``
+
+        * `if `kernel_config.is_not_set('FOO')``: ``if not kernel_config.get('FOO')``
+          (strictly: ``if kernel_config.get('FOO') is KernelConfigTristate.NO``)
+
+        * ``if kernel_config.is_module('FOO')``: ``if kernel_config.get('FOO') is KernelConfigTristate.MODULE``
```

Comment:
Since `.get()` and `[]` have a different semantic, it's a better idea to let the user choose which one is needed in a given context. If I test for `CONFIG_FOO` being set and this happens to not be a valid option at all, I'd rather have an exception raised than `is_not_set('CONFIG_FOO')` silently returning `False`. In the current form `not is_not_set('FOO')` does not imply that `FOO` is set, which is somehow misleading.

Also, being `is_enabled`, `is_module` and `is_not_set` only make sense for tristate options, which is modelled here by a type that already handles this check using `is` or `==`. Having `KernelConfigTristate` becomes necessary since in the previous API, tristate were "plain" strings and actual strings had an extra layer of double-quoting ('y' vs '"my string"'). It also allows `__bool__` to be defined which is handy and not possible on a 'y'/'n'/'m' string.

That said, this could be added if that use case is really frequent (same for the other similar methods):
```
    def is_not_set(self, name):
        return self.get(name) is not KernelConfigTristate.NO
```

Reply:
Fair enough, that makes sense and yes I agree it makes sense to raise an error if the desired option is not present all, I was just thinking that having something that includes `is_module` in the method name could be easier to interpret what the code was doing at a glance rather than `.get()` but yes it only makes sense in some contexts.